onnxruntime/core/platform/android/cxa_demangle.cc

```diff
@@ -0,0 +1,70 @@
+#if USE_DUMMY_EXA_DEMANGLE
+
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+
+#define DEMANGLE_INVALID_ARGUMENTS -3
+#define DEMANGLE_INVALID_MANGLED_NAME -2
+#define DEMANGLE_MEMORY_ALLOCATION_FAILURE -1
+#define DEMANGLE_SUCCESS 0
+
+// Reduced from https://github.com/llvm/llvm-project/blob/09c2b7c35a/libcxxabi/src/demangle/Utility.h
+inline bool initialize_output_buffer(char*& buf, size_t* n, size_t& buf_size, size_t init_size) {
+  // only handle buf == null case, is buf is provided, in original impl, it realloc to expand buf space.
```

Comment:
static

Reply:
pointer is only guarantee to be non-null iff buf is non-null